Blocstagram/CameraViewController.m

```diff
@@ -185,7 +185,8 @@ - (void) leftButtonPressedOnToolbar:(CameraToolbar *)toolbar
             newIndex = currentIndex + 1;
         }
         
-        AVCaptureDevice *newCamera = devices[newIndex];
+        AVCaptureDevice *newCamera = [[AVCaptureDevice alloc] init];
```

Comment:
You don't necessarily need this line... I think the old way worked fine... this way we just allocate an instance then immediately write over it.